codex-process-feasibility/src/test/java/de/netzwerk_universitaetsmedizin/codex/processes/feasibility/service/DownloadFeasibilityResourcesTest.java

```diff
@@ -0,0 +1,223 @@
+import static org.junit.Assert.assertThrows;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.StrictStubs.class)
+public class DownloadFeasibilityResourcesTest {
+
+    @Mock
+    private EnhancedFhirWebserviceClientProvider clientProvider;
+
+    @Mock
+    private FhirWebserviceClient webserviceClient;
+
+    @Mock
+    private TaskHelper taskHelper;
+
+    @Mock
+    private DelegateExecution execution;
+
+    @Mock
+    private Task task;
+
+    @InjectMocks
+    private DownloadFeasibilityResources service;
+
+    private Map<String, List<String>> createSearchQueryParts(String measureId) {
+        return Map.of("_id", Collections.singletonList(measureId), "_include",
+                Collections.singletonList("Measure:depends-on"));
+    }
+
+    @Test
+    public void testDoExecute_NoMeasureReference() {
+        when(execution.getVariable(BPMN_EXECUTION_VARIABLE_TASK)).thenReturn(task);
+        when(taskHelper.getFirstInputParameterReferenceValue(task, CODESYSTEM_FEASIBILITY,
+                CODESYSTEM_FEASIBILITY_VALUE_MEASURE_REFERENCE))
+                .thenReturn(Optional.empty());
+
+        assertThrows(RuntimeException.class, () -> service.execute(execution));
+        verify(task).setStatus(Task.TaskStatus.FAILED);
+    }
+
+    @Test
+    public void testDoExecute_BundleWithTooFewResultEntries() {
+        final String measureId = "id-151003";
+        final IdType measureRefId = new IdType("http://remote.host/Measure/" + measureId);
+        final Reference measureRef = new Reference(measureRefId);
+
+        when(execution.getVariable(BPMN_EXECUTION_VARIABLE_TASK))
+                .thenReturn(task);
+        when(taskHelper.getFirstInputParameterReferenceValue(task, CODESYSTEM_FEASIBILITY,
+                CODESYSTEM_FEASIBILITY_VALUE_MEASURE_REFERENCE))
+                .thenReturn(Optional.of(measureRef));
+        when(clientProvider.getWebserviceClient(measureRefId))
+                .thenReturn(webserviceClient);
+        when(webserviceClient.searchWithStrictHandling(Measure.class, createSearchQueryParts(measureId)))
+                .thenReturn(new Bundle());
+
+        assertThrows(RuntimeException.class, () -> service.execute(execution));
+        verify(task).setStatus(Task.TaskStatus.FAILED);
+    }
+
+    @Test
+    public void testDoExecute_FirstBundleEntryIsNoMeasure() {
+        final String measureId = "id-151003";
+        final IdType measureRefId = new IdType("http://remote.host/Measure/" + measureId);
+        final Reference measureRef = new Reference(measureRefId);
+
+        final Bundle.BundleEntryComponent patientEntryA = new Bundle.BundleEntryComponent()
+                .setResource(new Patient().setId("id-170524"));
+
+        final Bundle.BundleEntryComponent patientEntryB = new Bundle.BundleEntryComponent()
+                .setResource(new Patient().setId("id-123456"));
+
+        final Bundle measureOnlyBundle = new Bundle()
```

Comment:
```java
Bundle bundle = new Bundle();
bundle.addEntry().setResource(new Patient().setId("foo"));
bundle.addEntry().setResource(new Patient().setId("foo"));
```

is sufficient. I would not use a special ID here, because all it needs is a non-empty resource. Please replace all other ID's this way.